internal/store/plugin.go

```diff
@@ -29,19 +34,21 @@ func (store *Store) GetPlugins(pluginFilter *model.PluginFilter) ([]*model.Plugi
 		return nil, nil
 	}
 
-	var plugins []*model.Plugin
+	plugins, err := store.getPlugins(pluginFilter.ServerVersion)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get plugins")
+	}
 
 	filter := strings.TrimSpace(pluginFilter.Filter)
-	if filter == "" {
-		plugins = store.plugins
-	} else {
-		for _, plugin := range store.plugins {
-			if !pluginMatchesFilter(plugin, filter) {
-				continue
+	if filter != "" {
+		n := 0
+		for _, plugin := range plugins {
```

Comment:
This is clever, but I'm wondering if it's necessary? It takes an extra moment to read and parse and prove that it filters in place. Is there a pre-existing performance issue here?